### PR TITLE
[API Deprecation] Remove _deprecation module

### PR DESCRIPTION
**Pull request by peizhou001**
## Description
Delete entire **_deprecation** module and related functions.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by mufeili** [comment]
Have related people reviewed this before? Also make sure you address the CI failures of DGL-Go.

**Comment by peizhou001** [comment]
> Have related people reviewed this before? Also make sure you address the CI failures of DGL-Go.

Request @jermainewang  for review. For DGL-Go CI, I have fixed problem by upgrading OGB version to 1.3.5 locally, let me try it in remote.

File: python/dgl/function/message.py [hunk 1]
```diff
@@ -46,20 +44,7 @@ def __init__(self, binary_op, lhs, rhs, lhs_field, rhs_field, out_field):
 
     def _invoke(self, graph, src_frame, dst_frame, edge_frame, out_size,
```
Comment (jermainewang) [inline python/dgl/function/message.py:45]: The entire `_invoke` method could be removed.

File: python/dgl/function/reducer.py [hunk 2]
```diff
@@ -34,21 +32,7 @@ def __init__(self, name, msg_field, out_field):
         self.out_field = out_field
 
     def _invoke(self, graph, edge_frame, out_size, edge_map=None, out_map=None):
```
Comment (jermainewang) [inline python/dgl/function/reducer.py:34]: ditto

**Review by jermainewang — APPROVED** [review]

File: python/dgl/function/message.py (hunk 1 above)
Comment (peizhou001) [reply python/dgl/function/message.py]: fixed

File: python/dgl/function/reducer.py (hunk 2 above)
Comment (peizhou001) [reply python/dgl/function/reducer.py]: fixed

**Pull request merged**